processor/transformprocessor/config.go

```diff
@@ -44,6 +48,63 @@ type Config struct {
 	logger      *zap.Logger
 }
 
+// Unmarshal is used internally by mapstructure to parse the transformprocessor configuration (Config),
+// adding support to structured and flat configuration styles (array of statements strings).
+// When the flat configuration style is used, each statement becomes a new common.ContextStatements
+// object, with empty [common.ContextStatements.Context] value.
+// On the other hand, structured configurations are parsed following the mapstructure Config format.
+// Mixed configuration styles are also supported.
+func (c *Config) Unmarshal(component *confmap.Conf) error {
+	if component == nil {
+		return nil
+	}
+
+	contextStatementsPatch := map[string]any{}
+	for _, fieldName := range contextStatementsFields {
+		if !component.IsSet(fieldName) {
+			continue
+		}
+		rawVal := component.Get(fieldName)
+		values, ok := rawVal.([]any)
+		if !ok {
+			return fmt.Errorf("invalid %s type, expected: array, got: %t", fieldName, rawVal)
+		}
+		if len(values) == 0 {
+			continue
+		}
+
+		stmts := make([]any, 0, len(values))
+		for i, value := range values {
+			// Array of strings means it's a flat configuration style
+			if reflect.TypeOf(value).Kind() == reflect.String {
+				stmts = append(stmts, map[string]any{
+					"statements":   []any{value},
+					"shared_cache": true,
+				})
+			} else {
+				configuredKeys, ok := value.(map[string]any)
+				if ok {
+					_, hasShareCacheKey := configuredKeys["shared_cache"]
+					if hasShareCacheKey {
+						return fmt.Errorf("%s[%d] has invalid keys: %s", fieldName, i, "shared_cache")
+					}
+				}
```

Comment:
What is this check protecting against? Someone doing

```yaml
trace_statements:
  - name: span
    shared_cache: true
    statements: 
        - set(name, "test")
```

Reply:
Yes, exactly, although it wouldn't avoid API's consumers of setting/changing it.